bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/BookieRequestHandler.java

```diff
@@ -34,20 +40,24 @@
  */
 @Slf4j
 public class BookieRequestHandler extends ChannelInboundHandlerAdapter {
-
+    private static final int DEFAULT_CAPACITY = 1_000;
     static final Object EVENT_FLUSH_ALL_PENDING_RESPONSES = new Object();
 
     private final RequestProcessor requestProcessor;
     private final ChannelGroup allChannels;
 
     private ChannelHandlerContext ctx;
+    private final BlockingQueue<BookieProtocol.ParsedAddRequest> msgs;
 
     private ByteBuf pendingSendResponses = null;
     private int maxPendingResponsesSize;
 
     BookieRequestHandler(ServerConfiguration conf, RequestProcessor processor, ChannelGroup allChannels) {
         this.requestProcessor = processor;
         this.allChannels = allChannels;
+
+        int maxCapacity = conf.getMaxAddsInProgressLimit() > 0 ? conf.getMaxAddsInProgressLimit() : DEFAULT_CAPACITY;
+        this.msgs = new ArrayBlockingQueue<>(maxCapacity);
```

Comment:
I'm not sure if it's going to happen, but I wanted to ask this question: Do we need to add a byte size limit? How to prevent the channel from reading a large amount of data from the socket and causing the OOM of the direct memory?

Reply:
I think maybe not. The megs queue is also controlled by the Netty request handle. Netty also has a read batch control and once the batch limit is reached, it will call `channelReadComplete`, and we will flush the msgs queue requests to the storage when the `channelReadComplete` is called.

Reply:
I got it. Thanks!